### ISSUE TITLE
Cannot read property '_currentRetry' of undefined

### ISSUE THREAD

**Issue by dan-cooke** [cypress-io/cypress#8855]
### Current behavior
Running a smoke test suite with the following `cypress.json`

```json
{
  "projectId": "4bbr1p",
  "baseUrl": "http://127.0.0.1:5000/console/ui/",
  "defaultCommandTimeout": 180000,
  "env": {
    "codeCoverage": {
      "url": "/api/__coverage__"
    },
    "ELECTRON_ENABLE_LOGGING": 0,
    "USE_MOCKS": true
  },
  "requestTimeout": 180000,
  "responseTimeout": 180000,
  "retries": {
    "runMode": 2,
    "openMode": 0
  },
  "testFiles": "**/*.spec.js",
  "viewportHeight": 1080,
  "viewportWidth": 1920,
  "videoUploadOnPasses": false
}
```

We encounter the following error on our Travis CI - sporadically

```
TypeError: Cannot read property '_currentRetry' of undefined
    at Reporter.<anonymous> (/home/travis/.cache/Cypress/5.3.0/Cypress/resources/app/packages/server/lib/reporter.js:141:46)
    at Reporter.parseArgs (/home/travis/.cache/Cypress/5.3.0/Cypress/resources/app/packages/server/lib/reporter.js:337:18)
    at Reporter.emit (/home/travis/.cache/Cypress/5.3.0/Cypress/resources/app/packages/server/lib/reporter.js:319:17)
    at Object.onMocha (/home/travis/.cache/Cypress/5.3.0/Cypress/resources/app/packages/server/lib/project.js:360:18)
    at Socket.<anonymous> (/home/travis/.cache/Cypress/5.3.0/Cypress/resources/app/packages/server/lib/socket.js:320:32)
    at Socket.emit (events.js:223:5)
    at /home/travis/.cache/Cypress/5.3.0/Cypress/resources/app/packages/socket/node_modules/socket.io/lib/socket.js:528:12
```

It is unclear if the test had failed and attempted a retry.

### Desired behavior
Test should retry if failed, or continue on without crashing.

### Versions
Cypress 5.3.0

**Comment by jennifer-shehane** [comment]
Do you have a `test:before:run` defined in your testing suite somewhere? Can you provide the code within there?

Can you provide any test code to reproduce this? It's likely dependent on the tests you're running and its structure. 

The error is throwing here where the `runnable` is undefined: https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/reporter.js#L141:L141

**Comment by dan-cooke** [comment]
@jennifer-shehane 

We do not have a `test:before:run`, however we are using the Module API to run cypress programatically.
Apologies for no reproduction repo, but I cannot seem to reproduce on another repo, and I cannot share the repo I found this bug on.

Feel free to close the issue if there is not enough information

**Comment by jennifer-shehane** [comment]
Yeah, it would be great to track down, but can't really do much without a way to reproduce it. 

Please comment in this issue with a reproducible example and we will consider reopening the issue.

**Comment by flotwig** [comment]
Just got this in our CI: https://app.circleci.com/pipelines/github/cypress-io/cypress/27512/workflows/b160424c-80a0-42b5-99bd-8c13c2dd6427/jobs/1040757/parallel-runs/3?filterBy=FAILED

**Comment by mmonteiroc** [comment]
Any news on this issue, we just got the same error in our CI
We used for a while CHROME browser, and now we changed back to electron. 

Only once we changed back to electron, we got that issue
![image](https://user-images.githubusercontent.com/28816737/222673698-77a4722f-17e4-4da1-8224-1e6600748efc.png)

**Comment by SwardCoder** [comment]
Useful information to solve this issue 

Even when I was doing API testing I got the same issue and later upon investigating alot I found that 
The way I was testing my spec was like this in the form of Nested Describe and Iterations 
<pre>
describe("Main Describe", () => {

Before(() => { 
   // some initial code
}) 

it ("iteration1", () => { 
     // some assertions
   cy.request (.......) .then ( ....
          describe("Nested Describe", () => {
               it ("sub iteration1", () => { 
                    // some sub assertions
               }
          ) }
    ) }
})

it ("iteration2", () => { 
   // some assertions
})

})

</pre>

when i removed nested describe then i was not getting this issue so i would say please use alternatives like 
it or steps to get rid of this issue  if you find any nested describes/iterations in program

**Comment by jordanpowell88** [comment]
@SwardCoder I am not able to reproduce the issue you described above. My example below works as expected:

```ts
describe('page', () => {
  const api = 'https://place.dog/300/200';
  
  before(() => {
    cy.intercept(api).as('dog')
  })
  
  it('iteration 1', () => {
    cy.request(
      'GET',
      api
    ).then(({ status }) => {
      console.log(status)
      describe('Nested Describe', () => {
        expect(status).to.equal(200)
      })
    })
  })
})
```
![Nested Describe Cypress Output](https://user-images.githubusercontent.com/3605268/234078684-a8114086-3aae-4bd5-a533-47d0de24facd.png)

**Comment by warrensplayer** [comment]
@mmonteiroc I am closing this issue due to inactivity, but if you are still having this problem, please create a new issue and link to this one.
